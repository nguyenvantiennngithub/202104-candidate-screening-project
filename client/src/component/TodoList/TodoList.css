.todoList {
    border-bottom: 1px solid #363535;
}
.todoList-header {
    display: flex;
    justify-content: space-between;
    align-items: center;
}
