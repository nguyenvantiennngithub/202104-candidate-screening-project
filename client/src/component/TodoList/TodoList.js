import { Button } from "@mui/material";
import React from "react";
import { useDialog } from "../../hooks";
import { useTodo } from "../../store/hooks";
import TodoItem from "../TodoItem/TodoItem";
import DialogConfirm from "./Dialog/DialogConfirm";
import * as actionTodo from "../../store/todoAction";
import * as apiTodo from "../../api/todo";
import "./TodoList.css";

const TodoList = () => {
    const { state, dispatch } = useTodo();

    const { isOpenDialog, handleOpenDialog, handleCloseDialog } = useDialog();
    const handleClearCompleted = async () => {
        const listCompletedId = state.todos.reduce((total, todo) => {
            if (todo.isCompleted) total.push(todo.id);
            return total;
        }, []);

        const { data } = await apiTodo.clearCompleted({ listCompletedId });
        if (data.status === "success") {
            dispatch(actionTodo.clearCompleted(listCompletedId));
        }
    };
    return (
        <div className="todoList">
            <div className="todoList-header">
                <h2>Todo List</h2>
                <Button
                    color="error"
                    variant="contained"
                    onClick={handleOpenDialog}
                >
                    Clear complated
                </Button>
                <DialogConfirm
                    isOpenDialog={isOpenDialog}
                    handleCloseDialog={handleCloseDialog}
                    handleClearCompleted={handleClearCompleted}
                ></DialogConfirm>
            </div>
            {state.todos.map((item, index) => {
                return <TodoItem key={index} data={item}></TodoItem>;
            })}
        </div>
    );
};

export default TodoList;
