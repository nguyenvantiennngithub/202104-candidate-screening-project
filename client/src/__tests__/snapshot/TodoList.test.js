import TodoList from "../../component/TodoList/TodoList";
import Provider from "../../store/TodoProvider";
import renderer from "react-test-renderer";

describe("Todo list", () => {
    it("normal case", () => {
        const todoList = renderer.create(
            <Provider>
                <TodoList></TodoList>
            </Provider>
        );

        const todoListTree = todoList.toJSON();
        expect(todoListTree).toMatchSnapshot();
    });
});
