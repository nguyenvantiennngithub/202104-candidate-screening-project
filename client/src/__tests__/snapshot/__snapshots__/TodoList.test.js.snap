// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo list normal case 1`] = `
<div
  className="todoList"
>
  <div
    className="todoList-header"
  >
    <h2>
      Todo List
    </h2>
    <button
      className="MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-1sszw6i-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Clear complated
    </button>
  </div>
</div>
`;
